Vector layer scaling inconsistencies

## 🐛 Bug

When trying to use the `scale` parameter of vector layers, I encountered some issues. Type errors are raised unexpectedly, and arrays are reshaped strangely.

However, I can create a layer and *then* set its scale to have the expected behaviour.

EDIT: actually, it seems that using a single float is not accepted either way.

## To Reproduce

This works as intended:
```python
import napari
import numpy as np

v = napari.Viewer(ndisplay=3)
l = v.add_vectors(np.random.rand(10, 2, 3) * 100)
l.scale = [1, 2, 1]
```

However, this raises an error:
```python
l = v.add_vectors(np.random.rand(10, 2, 3) * 100, scale=[1, 2, 1])
.....
~/.local/lib/python3.9/site-packages/napari/utils/transforms/transforms.py in __call__(self, coords)
    347         if coords.shape[1] != self.linear_matrix.shape[0]:
    348             linear_matrix = np.eye(coords.shape[1])
--> 349             linear_matrix[
    350                 -self.linear_matrix.shape[0] :, -self.linear_matrix.shape[1] :
    351             ] = self.linear_matrix

ValueError: could not broadcast input array from shape (4,4) into shape (3,3)
```

The docs say that `scale` accepts a single float, but that fails as well with:
```python
TypeError: object of type 'int' has no len()
```

## Expected behavior

Same as setting `scale` posthumously.

## Environment

```yaml
napari: 0.4.6
Platform: Linux-5.11.2-arch1-1-x86_64-with-glibc2.33
System: Arch Linux
Python: 3.9.2 (default, Feb 20 2021, 18:40:11) [GCC 10.2.0]
Qt: 5.15.2
PyQt5: 5.15.2
NumPy: 1.20.1
SciPy: 1.6.1
Dask: 2020.12.0
VisPy: 0.6.6

OpenGL:
- GL version: 4.6.0 NVIDIA 460.56
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1920x1080, scale 1.0
- screen 2: resolution 1920x1200, scale 1.0

Plugins:
- console: 0.0.3
- mrcfile-reader: 0.1.2
- napari-em-reader: 0.1.0
- ndtiffs: 0.1.1
- svg: 0.1.4
```

Indeed, that's probably connected to the same inconsistency between setting attributes and calling `add_*`.
Actually, I found part of the same issue with the `Points` layer just now: using a single scalar does not work in the same way:
```python
E           ValueError: Input must be 1- or 2-d.
```

--- a/napari/layers/vectors/vectors.py
+++ b/napari/layers/vectors/vectors.py
@@ -203,8 +203,7 @@ class Vectors(Layer):
         # Save the vector style params
         self._edge_width = edge_width
 
-        # length attribute
-        self._length = length
+        self._length = float(length)
 
         self._data = data
 
@@ -409,7 +408,7 @@ class Vectors(Layer):
 
     @length.setter
     def length(self, length: Union[int, float]):
-        self._length = length
+        self._length = float(length)
 
         vertices, triangles = generate_vector_meshes(
             self.data[:, :, list(self._dims_displayed)],

